go/adbc/ext.go

```diff
@@ -159,3 +160,136 @@ func IngestStream(ctx context.Context, cnxn Connection, reader array.RecordReade
 
 	return count, nil
 }
+
+// VersionInfo contains comprehensive driver and library version information.
+type VersionInfo struct {
+	DriverName    string            `json:"driver_name"`    // e.g., "ADBC PostgreSQL Driver"
+	DriverVersion string            `json:"driver_version"` // e.g., "15.13.0000"
+	LibraryInfo   map[string]string `json:"library_info"`   // Additional library versions, protocol info, etc.
+}
+
+// GetVersionInfo retrieves comprehensive driver version information from a connection.
+// This helper function encapsulates the complex process of querying driver info codes,
+// handling streaming record batches and union arrays, extracting and safely cloning
+// string values, and managing errors with fallback defaults.
+//
+// Returns detailed version information including driver name, version, and additional
+// library information such as Arrow version, vendor details, and ADBC version.
+//
+// This is not part of the ADBC API specification.
+func GetVersionInfo(ctx context.Context, cnxn Connection) (*VersionInfo, error) {
+	stream, err := cnxn.GetInfo(ctx, []InfoCode{
+		InfoDriverName,
+		InfoDriverVersion,
+		InfoDriverArrowVersion,
+		InfoDriverADBCVersion,
+		InfoVendorName,
+		InfoVendorVersion,
+		InfoVendorArrowVersion,
+		InfoVendorSql,
+		InfoVendorSubstrait,
+		InfoVendorSubstraitMinVersion,
+		InfoVendorSubstraitMaxVersion,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error during GetInfo: %w", err)
+	}
+	defer stream.Release()
+
+	var name, version string
+	libInfo := make(map[string]string)
+
+	for stream.Next() {
+		batch := stream.Record()
+		defer batch.Release()
```

Comment:
this is unnecessary. If you don't call `Retain` on the record, then each call to `Next` will release the previous one